### DISCUSSION TITLE
Behaviour: refining an existing range selection in `RangeCalendar`?

### DISCUSSION THREAD

**Discussion by lunelson** [adobe/react-spectrum#6773] [Q&A]
I'm building a custom date range picker with RAC, and finding that once a range is selected, the state change that happens on the next click is somewhat counter-intuitive: namely, that the next click is simply treated as the start of a new range selection. 

In react-day-picker, which is the component basis I'm migrating away from in this case, after you create an initial range selection, the further clicks follow this logic:

- clicking a date before the `start` sets a new `start`, keeping the same `end` (extends the range)
- clicking a date after the `end` sets a new `end`, keeping the same `start` (extends the range)
- clicking a date between `start` and `end`, sets a new `end` (shortens the range)

One could debate the logic of the third case; but the first two cases are quite common, and intuitively implemented here—especially when the range spans multiple "pages" of the calendar, allowing you to modify the `start` or `end` without having to recreate the whole thing; so my question is: is there any way to get lower-level access to the range setting behaviour of `RangeCalendar`, so that I could allow refinement of an existing range rather than causing it to be cleared and forcing the user to start over?

**Comment by devongovett** [comment]
You can already drag the ends of a range to adjust them

**Reply by lunelson** [reply]
Thanks for that suggestion. Interesting feature; but TBH I don't think my users would discover it—and it wouldn't allow them pick day targets that are on the next or previous page. Ideally they'd be able to page to adjust the start value for example, by paging back by one month and then clicking again (keeping the endpoint retained). In my particular use-case in fact, because the focus is on time-spans in the past, it will be far more common for them to move the start point around but want to keep the endpoint (namely: today)
